irods_specific_implementation/irods_rest_api_base.h

```diff
@@ -132,6 +132,11 @@ namespace irods::rest
             load_client_api_plugins();
         } // ctor
 
+        inline bool http_bool(const std::string& s) noexcept
```

Comment:
If C strings are passed to this, it could result in a memory allocation. That can be avoided by replacing `std::string` with `std::string_view`.

Hence the following signature:
```cpp
bool http_bool(const std::string_view s) noexcept
```

Why is `inline` needed?

Reply:
Out of habit I just add inline to very short functions that don't call other functions. My understanding is that compilers these days will optimize most of these things better than a human anyway?

Reply:
Gotcha and you're absolutely right about compilers being really smart.